### PR TITLE
Support arrow functions in react-no-multiple-render-methods

### PR DESCRIPTION
**Pull request by vhpoet**
Title is self explanatory.

### CODE REVIEW

**Comment by GoodForOneFare** [comment]
Thanks for the contribution, Vahe! ❤️ @cartogram could you take a look when you have a moment, please?

**Comment by cartogram** [comment]
@vhpoet do mind adding a test for this?

File: lib/rules/react-no-multiple-render-methods.js [hunk 1]
```diff
@@ -47,10 +45,19 @@ module.exports = {
...
+
+        report(node);
+      },
     };
   }),
 };
 
-function isRenderMethod({key: {name}}) {
+function isRenderMethod(node) {
+  const name = (node.key && node.key.name) || (node.parent && node.parent.key && node.parent.key.name);
```
Comment (cartogram) [inline lib/rules/react-no-multiple-render-methods.js]: let's break this out into a small utility function. Something like:
```
getMethodName(node) {
  if (node.key) {
     return node.key.name
  }

  return node.parent && node.parent.key && node.parent.key.name;
}
```

File: lib/rules/react-no-multiple-render-methods.js [hunk 2]
```diff
@@ -47,10 +45,19 @@ module.exports = {
 
         report(node);
       },
+      
+      ArrowFunctionExpression(node) {
```
Comment (cartogram) [inline lib/rules/react-no-multiple-render-methods.js:49]: You are going to need to add a few tests for this. At the least I'd like to see it catch something like:
```
      code: `class Button extends React.Component {
        renderArrowFunction = () => {}
        renderBar = () => {}
        render() {}`
```

I think the error type will be different for these, so you will also need to modify the `error` utility to accept a `type` argument. Something like:

```
function error(memberName, type = 'MethodDefinition') {
  return {
    type,
    message: `Don’t use multiple render methods in a single component; they generally make your component harder to read. Instead break ${memberName} out into its own component and render it inside this one.`,
  };
}
```
then you should be able to do the following for the error of the above snippet:
```
errors: [
  error('renderArrowFunction', 'ArrowFunctionExpression'),
  error('renderBar', 'ArrowFunctionExpression'),
],
```

Happy to clarify anything you need.

**Review by cartogram — CHANGES_REQUESTED** [review]
Thanks for this contribution! Will take a look again once these changes are in 🙂

**Comment by alexandcote** [comment]
I did the rework you asked @cartogram. Let me know if everything is 👍🏼.

Thanks again @vhpoet 🤟🏼

**Review by BPScott — APPROVED** [review]
Not tried this but the test LGTM

**Pull request merged**